src/clj/game/core/engine.clj

```diff
@@ -1053,6 +1065,46 @@
                 (effect-completed state nil eid))
       (effect-completed state nil eid))))
 
+(defn trash-on-tag
+  [state _ eid]
+  (let [trash-when-tagged (when (jinteki.utils/is-tagged? state)
+                            (filter :trash-when-tagged (all-installed-runner state)))
+        trash-when-tagged (filter #(not (is-disabled? state nil %)) trash-when-tagged)]
+    (if (seq trash-when-tagged)
+      (wait-for (move* state nil (make-eid state eid)
+                       :trash-cards trash-when-tagged
+                       {:unpreventable true})
+                (doseq [card trash-when-tagged]
+                  (system-say state (to-keyword (:side card))
+                              (str "trashes " (card-str state card) " for being tagged")))
+                (effect-completed state nil eid))
+      (effect-completed state nil eid))))
+
+(defn- enforce-conditions-impl
+  [state _ eid cards]
+  (if (seq cards)
+    (let [fc (first cards)
+          rest (rest cards)]
+      (wait-for
+        (resolve-ability
+          state (to-keyword (:side fc))
+          (when-not
+              (is-disabled-reg? state fc)
+            (:enforce-conditions fc))
+          fc nil)
+        (enforce-conditions-impl state nil eid rest)))
+    (effect-completed state nil eid)))
+
+(defn enforce-conditions
+  [state _ eid]
+  (wait-for
+    (trash-on-tag state nil (make-eid state eid))
+    (if-let [cards (filter
```

Comment:
This will always be true because even an empty sequence is considered truthy. You gotta call `seq` to coerce to `nil` when empty: `(->> [(get-in ...)] (filter ...) (seq))`.